packages/client-react-streaming/src/DataTransportAbstraction/WrappedApolloClient.tsx

```diff
@@ -334,123 +247,71 @@ export function skipDataTransport<T extends Record<string, any>>(
+      !(options.context as InternalContext | undefined)?.[skipDataTransportKey]
     ) {
-      // don't transport the same query over twice
-      this.forwardedQueries.lookupArray(cacheKeyArr);
-      const observableQuery = super.watchQuery(options);
-      const queryInfo = observableQuery["queryInfo"] as QueryInfo;
-      const id = queryInfo.queryId as TransportIdentifier;
-
-      const streamObservable = new Observable<
-        Exclude<QueryEvent, { type: "started" }>
-      >((subscriber) => {
-        const { markResult, markError, markReady } = queryInfo;
-        queryInfo.markResult = function (result: FetchResult<any>) {
-          subscriber.next({
-            type: "data",
-            id,
-            result,
-          });
-          return markResult.apply(queryInfo, arguments as any);
-        };
-        queryInfo.markError = function () {
-          subscriber.next({
-            type: "error",
-            id,
-          });
-          subscriber.complete();
-          return markError.apply(queryInfo, arguments as any);
-        };
-        queryInfo.markReady = function () {
-          subscriber.next({
-            type: "complete",
-            id,
-          });
-          subscriber.complete();
-          return markReady.apply(queryInfo, arguments as any);
-        };
-      });
-
-      this.watchQueryQueue.push({
-        event: {
-          type: "started",
-          options: serializeOptions(options),
-          id,
-        },
-        observable: streamObservable,
+      return super.watchQuery({
+        ...options,
+        context: teeToReadableStream(() => this.pushEventStream(options), {
+          ...options?.context,
+        }),
       });
-      return observableQuery;
     }
     return super.watchQuery(options);
   }
-
-  onQueryStarted(event: Extract<QueryEvent, { type: "started" }>) {
-    const hydratedOptions = deserializeOptions(event.options);
-    const { cacheKeyArr } = this.identifyUniqueQuery(hydratedOptions);
-    // this is a replay from another source and doesn't need to be transported
-    // to the browser, since it will be replayed there, too.
-    this.forwardedQueries.lookupArray(cacheKeyArr);
-    super.onQueryStarted(event);
-  }
 }
 
-export class ApolloClientBrowserImpl extends ApolloClientClientBaseImpl {
-  constructor(options: WrappedApolloClientOptions) {
-    super(options);
-    this.setLink(this.link);
-  }
+class ApolloClientBrowserImpl extends ApolloClientClientBaseImpl {}
 
-  setLink(newLink: ApolloLink) {
-    super.setLink.call(this, new ReadFromReadableStreamLink().concat(newLink));
-  }
-}
-
-class ApolloClientRSCImpl extends ApolloClientBase {
-  constructor(options: WrappedApolloClientOptions) {
-    super(options);
-    this.setLink(this.link);
-  }
-
-  setLink(newLink: ApolloLink) {
-    super.setLink.call(this, new TeeToReadableStreamLink().concat(newLink));
-  }
-}
+class ApolloClientRSCImpl extends ApolloClientBase {}
```

Comment:
Leaving these here even though they're empty - those class names might help in stack traces submitted by users to more quickly identify problems (and tbh., at some point I'll add back some env-specific logic here anyways)